CMakeLists.txt

```diff
@@ -2,31 +2,29 @@ cmake_minimum_required(VERSION 3.3)
 cmake_policy(VERSION 3.3)
 
 # get version from Python package
-find_package(PythonInterp REQUIRED)
-execute_process(COMMAND ${PYTHON_EXECUTABLE} setup.py --version
-                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-                RESULT_VARIABLE PY_VERSION_FAILED
-                OUTPUT_VARIABLE PY_PKG_VERSION_FULL
-                ERROR_VARIABLE PY_VERSION_ERROR
+execute_process(COMMAND git describe --tags
```

Comment:
how does this work when one is between versions?
Also, are tags guaranteed to be sorted by date?

Reply:
From the docs:
```
The command finds the most recent tag that is reachable from a commit.
```

The current implementation also strips any additional information besides the last commit. CMake does not seem to understand the concept of a development version?

Reply:
> From the docs

Sadly, git is a moving target, and I've they don't really guarantee stability except for --porcelain commands.  I have the scars to prove it :(

CMake has that deficiency afaik - at least on the python side we can call things `.dev0`

Reply:
I've done some development by Travis… locally checking that CMake does reasonable stuff. In the last incarnation, `git` should not be called when building with `setup.py`, and could be overwritten by the user.

Should we make the pure C++ build fool-proof, too? Since one could just add `-DSONATA_VERSION`, I feel like the C++ build is reasonable.

Reply:
That seems like nice compromise.

I'm going to add @tomdele @wizmer to the review, so they are in the loop.